### PR TITLE
Updated for the new gfx-0.5

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

**Comment by mitchmindtree** [comment]
Thanks!

File: src/lib.rs [hunk 1]
```diff
@@ -58,18 +66,20 @@ impl<W, T> PistonWindow<W, T>
         use piston::event::Events;
         use piston::window::{ OpenGLWindow, Window };
 
-        let (mut device, mut factory) = gfx_device_gl::create(|s| window.borrow_mut().get_proc_address(s));
+        let mut device = gfx_device_gl::Device::new(|s| window.borrow_mut().get_proc_address(s));
+        let mut factory = device.spawn_factory();
```
Comment (mitchmindtree) [inline src/lib.rs]: @kvark should this factory be stored within the PistonWindow as well, now that `Canvas` has been removed? Or should it no longer be necessary?

Comment (mitchmindtree) [inline src/lib.rs]: Hmmm I thought that adding factory as a member of `PistonWindow` might then make it available for use with gfx_graphics' [`GlyphCache::new`](https://github.com/PistonDevelopers/gfx_graphics/blob/master/src/glyph.rs#L44) and [`GlyphCache::update`](https://github.com/PistonDevelopers/gfx_graphics/blob/master/src/glyph.rs#L58) method, but it seems that even if I add it, I still get the following error when trying to use it:
```
error: the trait `gfx::device::Factory<_>` is not implemented for the type `&mut gfx_device_gl::factory::Factory`
let mut glyph_cache = GlyphCache::new(&font_path, &mut window.factory.borrow_mut().deref_mut()).unwrap();
```
Perhaps the gfx_graphics and gfx_texture APIs need some changes to catch up with the latest gfx-rs version?

Comment (kvark) [inline src/lib.rs]: @mitchmindtree `draw_3d` is the only function where factory was really used (as a part of `Canvas`). The general idea behind multiple factories is that anything that manages its own resources should just carry its own factory (thus `draw*` is not supposed to use the factory parameter). `GlyphCash` is a good example for it, it just needs to be updated.

Comment (kvark) [inline src/lib.rs]: @mitchmindtree I believe it's going to work fine now after https://github.com/PistonDevelopers/gfx_graphics/pull/241 . Please let me know if you still have issues after it's merged.

**Pull request merged**